lib/rubocop/cop/rspec/support/aggregate_examples/matchers_with_side_effects.rb

```diff
@@ -0,0 +1,80 @@
+module RuboCop
+  module Cop
+    module RSpec
+      class AggregateExamples < Cop
+        # When aggregated, the expectations will fail when not supposed to or
+        # have a risk of not failing when expected to. One example is
+        # `validate_presence_of :comment` as it leaves an empty comment after
+        # itself on the subject making it invalid and the subsequent expectation
+        # to fail.
+        # Examples with those matchers are not supposed to be aggregated.
+        #
+        # @example MatchersWithSideEffects
+        #
+        #   # .rubocop.yml
+        #   # RSpec/AggregateExamples:
+        #   #   MatchersWithSideEffects:
+        #   #   - allow_value
+        #   #   - allow_values
+        #   #   - validate_presence_of
+        #
+        #   # bad, but isn't automatically correctable
+        #   describe do
+        #     it { is_expected.to validate_presence_of(:comment) }
+        #     it { is_expected.to be_valid }
+        #   end
+        #
+        # @internal
+        #   Support for taking special care of the matchers that have side
+        #   effects, i.e. leave the subject in a modified stte.
+        module MatchersWithSideEffects
```

Comment:
I'm trying to reduce the scope of this pull request, and one of the things I plan to extract is a `MatcherWithSideEffects` cop that would prevent from using the matchers that leave the dirty subject after them together with the other matchers.

However, if I would remove this part from `AggregateExamples`, and it will suggest combining several examples into one, but `MatcherWithSideEffects` will suggest separating them.
I don't see that those two cops are in any way mutually exclusive in general, but this how would `AggregateExamples` know which examples to ignore?

@dgollahon You had a perspective on cop responsibilities separation, what do you think about this case?